python/dask_cudf/dask_cudf/tests/test_groupby.py

```diff
@@ -265,6 +264,46 @@ def test_groupby_dropna(dropna, by):
     dd.assert_eq(dask_result, cudf_result)
 
 
+@pytest.mark.parametrize("dropna", [False, True, None])
+@pytest.mark.parametrize(
+    "by", ["a", "b", "c", "d", ["a", "b"], ["a", "c"], ["a", "d"]]
+)
+def test_groupby_dropna_dask(dropna, by):
+    # see https://github.com/dask/dask/issues/8817
+    if len(by) > 1 and dropna is False:
+        pytest.skip("Multi-column dropna support is broken for Dask CPU")
+    if by == "d" or "d" in by:
+        pytest.skip(
+            "Dask CPU doesn't have support for dropna with categorical columns"
+        )
```

Comment:
What is the reason to include those "by" parameters if we will always be skipping them?  Where these tests only meant to be skipped when `dropna=True`?

Reply:
Good catch - IIRC they don't work when `dropna=False`, but I will try things out locally to double check

Reply:
Running the tests locally, it looks like there's a few separate (but likely related) issues in Dask CPU going on:

- multi-column groupbys grouping on categorical columns don't work at all
- `dropna=False` doesn't work on any multi-column groupbys
- `dropna=False` doesn't work on single-column groupbys on categorical columns

I think maybe instead of skipping here, we should xfail the currently failing cases - I suggest this so we can track the ongoing work in Dask CPU and remove the xfail once these tests are passing, without having to substantially change the test parameter. How do you feel about this? If you would rather just remove the categorical column cases for now and add a `TODO` to add those in later on when the dask issue is resolved, I am okay with that 🙂

Reply:
Sorry for mising this 26 days ago @charlesbluca !  I do think xfail makes a lot more sense than skipping.